renderdoc/driver/gl/gl_pixelhistory.cpp

```diff
@@ -250,9 +398,48 @@ rdcarray<EventUsage> QueryModifyingEvents(WrappedOpenGL *driver, GLPixelHistoryR
   return modEvents;
 }
 
+void QueryPostModPixelValues(WrappedOpenGL *driver, GLPixelHistoryResources &resources,
+                             const rdcarray<EventUsage> &modEvents, int x, int y,
+                             rdcarray<PixelModification> &history)
+{
+  driver->glBindFramebuffer(eGL_FRAMEBUFFER, resources.frameBuffer);
+  driver->glClear(eGL_COLOR_BUFFER_BIT | eGL_DEPTH_BUFFER_BIT | eGL_STENCIL_BUFFER_BIT);
+  driver->ReplayLog(0, modEvents[0].eventId, eReplay_WithoutDraw);
+
+  for(size_t i = 0; i < modEvents.size(); i++)
+  {
+    GLint savedReadFramebuffer, savedDrawFramebuffer;
+    driver->glGetIntegerv(eGL_DRAW_FRAMEBUFFER_BINDING, &savedDrawFramebuffer);
+    driver->glGetIntegerv(eGL_READ_FRAMEBUFFER_BINDING, &savedReadFramebuffer);
+    // bind our own framebuffer to save the pixel values
+    driver->glBindFramebuffer(eGL_FRAMEBUFFER, resources.frameBuffer);
+    driver->ReplayLog(modEvents[i].eventId, modEvents[i].eventId, eReplay_Full);
```

Comment:
Using `eReplay_Full` here is not the intended use for replaying a single draw. In practice currently if you specify the same EIDs for start and end it will be equivalent, but generally `eReplay_Full` and `eReplay_WithoutDraw` are reserved for replaying ranges of events (either an half-closed or entirely closed interval, effectively), with `eReplay_OnlyDraw` for replaying a single draw.

I admit this is confusing, especially when there's redundancy between the options and which EIDs you specify - because this API is internal it's been unchanged pretty much since the first implementation. It might merit a refactor in future to be more clear but that's not a priority at the moment.